From 0a60140df7bbbd5fdbb1555f21009f4100716308 Mon Sep 17 00:00:00 2001
From: pkandiX <pradeep.kumarx.kandi@intel.com>
Date: Tue, 8 Jan 2013 15:52:01 +0530
Subject: [PORT FROM MAIN]Game crashes when played with audio on

BZ: 76736

Checking for duration if it is greater than zero and then sending END_OF_STREAM error if required.

Change-Id: I84599666bed07bdb1e5d6f5cdcc1613be70edd1b
Orig-Change-Id: Ie93194019aa6d4d5b4f44572ea75815ce3061836
Signed-off-by: pkandiX <pradeep.kumarx.kandi@intel.com>
---
 media/libstagefright/OggExtractor.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/OggExtractor.cpp b/media/libstagefright/OggExtractor.cpp
index 2cafa3a..8ddf120 100644
--- a/media/libstagefright/OggExtractor.cpp
+++ b/media/libstagefright/OggExtractor.cpp
@@ -507,7 +507,7 @@ status_t MyVorbisExtractor::readNextPacket(MediaBuffer **out) {
             if (timeUs > 0) {
                 int64_t durationUs = 0;
                 if (mMeta->findInt64(kKeyDuration, &durationUs)) {
-                    if (durationUs < timeUs) {
+                    if (durationUs < timeUs && durationUs > 0) {
                         LOGV("Reached End of File");
                         return ERROR_END_OF_STREAM;
                     }
-- 
1.7.4.1

